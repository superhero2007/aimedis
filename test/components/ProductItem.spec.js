import React from 'react';
import { mount } from 'enzyme';

import ProductItem from 'components/ProductItem';

function setup(align, description = 'Description', src = 'Product1.png', title = 'Title') {
  return mount(<ProductItem
    align={align}
    description={description}
    src={src}
    title={title}
  />);
}

describe('ProductItem', () => {
  const leftWrapper = setup('left');

  it('should render properly', () => {
    expect(leftWrapper.html()).toMatchSnapshot();
  });

  const rightWrapper = setup('right');

  it('should render properly', () => {
    expect(rightWrapper.html()).toMatchSnapshot();
  });
});
