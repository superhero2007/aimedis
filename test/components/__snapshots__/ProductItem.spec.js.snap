// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem should render properly 1`] = `"<div class=\\"app__product__item\\"><div class=\\"app__product__item__content\\"><div class=\\"row\\"><div class=\\"col-12 col-lg-6 p-lg-0 app__product__item__content__info\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div><p class=\\"app__product__item__content__info__title\\">Title</p><div class=\\"app__product__item__content__info__image d-block d-lg-none\\"><img src=\\"\\" alt=\\"No Asset\\"></div><p class=\\"app__product__item__content__info__description\\">Description</p><div class=\\"app__product__item__content__info__button\\"><div class=\\"app__product__item__content__info__button__icon\\"><div class=\\"app__product__item__content__info__button__icon__triangle\\"></div></div><p class=\\"app__product__item__content__info__button__text\\">PLAY VIDEO</p></div></div></div></div><div class=\\"col-6 d-none d-lg-flex\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><img src=\\"\\" alt=\\"No Asset\\"></div></div></div></div><div class=\\" d-none\\"><div class=\\"app__product__item__content__player\\"><div class=\\"video-react-controls-enabled video-react-paused video-react-fluid video-react-user-active video-react-workinghover video-react\\" style=\\"padding-top: 56.25%;\\" role=\\"region\\" tabindex=\\"-1\\"><video class=\\"video-react-video\\" preload=\\"auto\\" tabindex=\\"-1\\"></video><div class=\\"video-react-loading-spinner\\"></div><button class=\\"video-react-big-play-button video-react-big-play-button-left big-play-button-hide\\" type=\\"button\\" aria-live=\\"polite\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play Video</span></button><div class=\\"video-react-control-bar video-react-control-bar-auto-hide\\"><button class=\\"video-react-play-control video-react-control video-react-button video-react-paused\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play</span></button><div class=\\"video-react-volume-menu-button-horizontal video-react-vol-3 video-react-volume-menu-button video-react-menu-button-inline video-react-control video-react-button video-react-menu-button\\" role=\\"button\\" tabindex=\\"0\\"><div class=\\"video-react-menu\\"><div class=\\"video-react-menu-content\\"><div class=\\"video-react-volume-bar video-react-slider-bar video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"volume level\\" aria-valuenow=\\"100.00\\" aria-valuetext=\\"100.00%\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div class=\\"video-react-volume-level\\" style=\\"width: 100.00%;\\"><span class=\\"video-react-control-text\\"></span></div></div></div></div></div><div class=\\"video-react-current-time video-react-time-control video-react-control\\"><div class=\\"video-react-current-time-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Current Time </span>0:00</div></div><div class=\\"video-react-time-control video-react-time-divider\\" dir=\\"ltr\\"><div><span>/</span></div></div><div class=\\"video-react-duration video-react-time-control video-react-control\\"><div class=\\"video-react-duration-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Duration Time </span>0:00</div></div><div class=\\"video-react-progress-control video-react-control\\"><div class=\\"video-react-progress-holder video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"video progress bar\\" aria-valuenow=\\"NaN\\" aria-valuetext=\\"0:00\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div data-current-time=\\"0:00\\" class=\\"video-react-play-progress video-react-slider-bar\\"><span class=\\"video-react-control-text\\">Progress: NaN%</span></div></div></div><button class=\\"video-react-icon-fullscreen video-react-fullscreen-control video-react-control video-react-button video-react-icon\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Non-Fullscreen</span></button></div></div><div class=\\"app__product__item__content__player__close\\"><i class=\\"i-close\\"></i></div></div></div></div>"`;

exports[`ProductItem should render properly 2`] = `"<div class=\\"app__product__item\\"><div class=\\"app__product__item__content\\"><div class=\\"row\\"><div class=\\"col-6 d-none d-lg-flex direction-rtl\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><img src=\\"\\" alt=\\"No Asset\\"></div></div><div class=\\"col-12 col-lg-6 p-lg-0 app__product__item__content__info\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div><p class=\\"app__product__item__content__info__title\\">Title</p><div class=\\"app__product__item__content__info__image d-block d-lg-none\\"><img src=\\"\\" alt=\\"No Asset\\"></div><p class=\\"app__product__item__content__info__description\\">Description</p><div class=\\"app__product__item__content__info__button\\"><div class=\\"app__product__item__content__info__button__icon\\"><div class=\\"app__product__item__content__info__button__icon__triangle\\"></div></div><p class=\\"app__product__item__content__info__button__text\\">PLAY VIDEO</p></div></div></div></div></div></div><div class=\\" d-none\\"><div class=\\"app__product__item__content__player\\"><div class=\\"video-react-controls-enabled video-react-paused video-react-fluid video-react-user-active video-react-workinghover video-react\\" style=\\"padding-top: 56.25%;\\" role=\\"region\\" tabindex=\\"-1\\"><video class=\\"video-react-video\\" preload=\\"auto\\" tabindex=\\"-1\\"></video><div class=\\"video-react-loading-spinner\\"></div><button class=\\"video-react-big-play-button video-react-big-play-button-left big-play-button-hide\\" type=\\"button\\" aria-live=\\"polite\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play Video</span></button><div class=\\"video-react-control-bar video-react-control-bar-auto-hide\\"><button class=\\"video-react-play-control video-react-control video-react-button video-react-paused\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play</span></button><div class=\\"video-react-volume-menu-button-horizontal video-react-vol-3 video-react-volume-menu-button video-react-menu-button-inline video-react-control video-react-button video-react-menu-button\\" role=\\"button\\" tabindex=\\"0\\"><div class=\\"video-react-menu\\"><div class=\\"video-react-menu-content\\"><div class=\\"video-react-volume-bar video-react-slider-bar video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"volume level\\" aria-valuenow=\\"100.00\\" aria-valuetext=\\"100.00%\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div class=\\"video-react-volume-level\\" style=\\"width: 100.00%;\\"><span class=\\"video-react-control-text\\"></span></div></div></div></div></div><div class=\\"video-react-current-time video-react-time-control video-react-control\\"><div class=\\"video-react-current-time-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Current Time </span>0:00</div></div><div class=\\"video-react-time-control video-react-time-divider\\" dir=\\"ltr\\"><div><span>/</span></div></div><div class=\\"video-react-duration video-react-time-control video-react-control\\"><div class=\\"video-react-duration-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Duration Time </span>0:00</div></div><div class=\\"video-react-progress-control video-react-control\\"><div class=\\"video-react-progress-holder video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"video progress bar\\" aria-valuenow=\\"NaN\\" aria-valuetext=\\"0:00\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div data-current-time=\\"0:00\\" class=\\"video-react-play-progress video-react-slider-bar\\"><span class=\\"video-react-control-text\\">Progress: NaN%</span></div></div></div><button class=\\"video-react-icon-fullscreen video-react-fullscreen-control video-react-control video-react-button video-react-icon\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Non-Fullscreen</span></button></div></div><div class=\\"app__product__item__content__player__close\\"><i class=\\"i-close\\"></i></div></div></div></div>"`;

exports[`ProductItem should render properly 3`] = `"<div class=\\"app__product__item\\"><div class=\\"app__product__item__content\\"><div class=\\"row\\"><div class=\\"col-6 d-none d-lg-flex direction-rtl\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><img src=\\"\\" alt=\\"No Asset\\"></div></div><div class=\\"col-12 col-lg-6 p-lg-0 app__product__item__content__info\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div><p class=\\"app__product__item__content__info__title\\">Title</p><div class=\\"app__product__item__content__info__image d-block d-lg-none\\"><img src=\\"\\" alt=\\"No Asset\\"></div><p class=\\"app__product__item__content__info__description\\">Description</p><div class=\\"app__product__item__content__info__button\\"><div class=\\"app__product__item__content__info__button__icon\\"><div class=\\"app__product__item__content__info__button__icon__triangle\\"></div></div><p class=\\"app__product__item__content__info__button__text\\">PLAY VIDEO</p></div></div></div></div></div></div><div class=\\" d-flex\\"><div class=\\"app__product__item__content__player\\"><div class=\\"video-react-controls-enabled video-react-paused video-react-fluid video-react-user-active video-react-workinghover video-react\\" style=\\"padding-top: 56.25%;\\" role=\\"region\\" tabindex=\\"-1\\"><video class=\\"video-react-video\\" preload=\\"auto\\" tabindex=\\"-1\\"></video><div class=\\"video-react-loading-spinner\\"></div><button class=\\"video-react-big-play-button video-react-big-play-button-left big-play-button-hide\\" type=\\"button\\" aria-live=\\"polite\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play Video</span></button><div class=\\"video-react-control-bar video-react-control-bar-auto-hide\\"><button class=\\"video-react-play-control video-react-control video-react-button video-react-paused\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play</span></button><div class=\\"video-react-volume-menu-button-horizontal video-react-vol-3 video-react-volume-menu-button video-react-menu-button-inline video-react-control video-react-button video-react-menu-button\\" role=\\"button\\" tabindex=\\"0\\"><div class=\\"video-react-menu\\"><div class=\\"video-react-menu-content\\"><div class=\\"video-react-volume-bar video-react-slider-bar video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"volume level\\" aria-valuenow=\\"100.00\\" aria-valuetext=\\"100.00%\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div class=\\"video-react-volume-level\\" style=\\"width: 100.00%;\\"><span class=\\"video-react-control-text\\"></span></div></div></div></div></div><div class=\\"video-react-current-time video-react-time-control video-react-control\\"><div class=\\"video-react-current-time-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Current Time </span>0:00</div></div><div class=\\"video-react-time-control video-react-time-divider\\" dir=\\"ltr\\"><div><span>/</span></div></div><div class=\\"video-react-duration video-react-time-control video-react-control\\"><div class=\\"video-react-duration-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Duration Time </span>0:00</div></div><div class=\\"video-react-progress-control video-react-control\\"><div class=\\"video-react-progress-holder video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"video progress bar\\" aria-valuenow=\\"NaN\\" aria-valuetext=\\"0:00\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div data-current-time=\\"0:00\\" class=\\"video-react-play-progress video-react-slider-bar\\"><span class=\\"video-react-control-text\\">Progress: NaN%</span></div></div></div><button class=\\"video-react-icon-fullscreen video-react-fullscreen-control video-react-control video-react-button video-react-icon\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Non-Fullscreen</span></button></div></div><div class=\\"app__product__item__content__player__close\\"><i class=\\"i-close\\"></i></div></div></div></div>"`;

exports[`ProductItem should render properly 4`] = `"<div class=\\"app__product__item\\"><div class=\\"app__product__item__content\\"><div class=\\"row\\"><div class=\\"col-6 d-none d-lg-flex direction-rtl\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><img src=\\"\\" alt=\\"No Asset\\"></div></div><div class=\\"col-12 col-lg-6 p-lg-0 app__product__item__content__info\\"><div class=\\"animated\\" style=\\"animation-duration: 1s; opacity: 0;\\"><div><p class=\\"app__product__item__content__info__title\\">Title</p><div class=\\"app__product__item__content__info__image d-block d-lg-none\\"><img src=\\"\\" alt=\\"No Asset\\"></div><p class=\\"app__product__item__content__info__description\\">Description</p><div class=\\"app__product__item__content__info__button\\"><div class=\\"app__product__item__content__info__button__icon\\"><div class=\\"app__product__item__content__info__button__icon__triangle\\"></div></div><p class=\\"app__product__item__content__info__button__text\\">PLAY VIDEO</p></div></div></div></div></div></div><div class=\\" d-none\\"><div class=\\"app__product__item__content__player\\"><div class=\\"video-react-controls-enabled video-react-paused video-react-fluid video-react-user-active video-react-workinghover video-react\\" style=\\"padding-top: 56.25%;\\" role=\\"region\\" tabindex=\\"-1\\"><video class=\\"video-react-video\\" preload=\\"auto\\" tabindex=\\"-1\\"></video><div class=\\"video-react-loading-spinner\\"></div><button class=\\"video-react-big-play-button video-react-big-play-button-left big-play-button-hide\\" type=\\"button\\" aria-live=\\"polite\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play Video</span></button><div class=\\"video-react-control-bar video-react-control-bar-auto-hide\\"><button class=\\"video-react-play-control video-react-control video-react-button video-react-paused\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Play</span></button><div class=\\"video-react-volume-menu-button-horizontal video-react-vol-3 video-react-volume-menu-button video-react-menu-button-inline video-react-control video-react-button video-react-menu-button\\" role=\\"button\\" tabindex=\\"0\\"><div class=\\"video-react-menu\\"><div class=\\"video-react-menu-content\\"><div class=\\"video-react-volume-bar video-react-slider-bar video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"volume level\\" aria-valuenow=\\"100.00\\" aria-valuetext=\\"100.00%\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div class=\\"video-react-volume-level\\" style=\\"width: 100.00%;\\"><span class=\\"video-react-control-text\\"></span></div></div></div></div></div><div class=\\"video-react-current-time video-react-time-control video-react-control\\"><div class=\\"video-react-current-time-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Current Time </span>0:00</div></div><div class=\\"video-react-time-control video-react-time-divider\\" dir=\\"ltr\\"><div><span>/</span></div></div><div class=\\"video-react-duration video-react-time-control video-react-control\\"><div class=\\"video-react-duration-display\\" aria-live=\\"off\\"><span class=\\"video-react-control-text\\">Duration Time </span>0:00</div></div><div class=\\"video-react-progress-control video-react-control\\"><div class=\\"video-react-progress-holder video-react-slider-horizontal video-react-slider\\" tabindex=\\"0\\" role=\\"slider\\" aria-label=\\"video progress bar\\" aria-valuenow=\\"NaN\\" aria-valuetext=\\"0:00\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\"><div data-current-time=\\"0:00\\" class=\\"video-react-play-progress video-react-slider-bar\\"><span class=\\"video-react-control-text\\">Progress: NaN%</span></div></div></div><button class=\\"video-react-icon-fullscreen video-react-fullscreen-control video-react-control video-react-button video-react-icon\\" type=\\"button\\" tabindex=\\"0\\"><span class=\\"video-react-control-text\\">Non-Fullscreen</span></button></div></div><div class=\\"app__product__item__content__player__close\\"><i class=\\"i-close\\"></i></div></div></div></div>"`;
